hal/src/nRF52840/exflash_hal.c

```diff
@@ -487,20 +695,11 @@ int hal_exflash_special_command(hal_exflash_special_sector_t sp, hal_exflash_com
             if (ret) {
                 goto exit;
             }
-            //Switch to qpsi mode
-#if HAL_PLATFORM_FLASH_MX25R6435FZNIL0
-            uint8_t temporary[3] = { 0x40, 0x00, 0x02 };
-            cinstr_cfg.opcode = QSPI_STD_CMD_WRSR;
-            cinstr_cfg.length = NRF_QSPI_CINSTR_LEN_4B;
-            ret = exflash_qspi_cinstr_xfer(&cinstr_cfg, temporary, NULL);
-#elif HAL_PLATFORM_FLASH_MX25L3233F
-            uint8_t temporary[1] = { 0x40 };
-            cinstr_cfg.opcode = QSPI_STD_CMD_WRSR;
-            cinstr_cfg.length = NRF_QSPI_CINSTR_LEN_2B;
-            ret = exflash_qspi_cinstr_xfer(&cinstr_cfg, temporary, NULL);
-#else
-#error "Unsupported platform for external flash"
-#endif
+
+            if (hal_exflash_driver.init) {
+                // TODO: Consolidate init here?
+                ret = hal_exflash_driver.init();
```

Comment:
The actual flash part init procedures aren't that complicated. We could just do an if/else here and do it inline rather than this func pointer if thats easier.